44385 Improve documentation on customizing the config screen (#345)
replicatedhq/replicated-docs: docs/vendor/config-screen-about.md

# About the Configuration Screen

> Applications can include a configuration screen in the Replicated admin console to collect required or optional values from your users that are used to start your application.

Reviewer: ```suggestion
Applications can include a configuration screen in the Replicated admin console to collect required or optional values from your users that are used to run your application.
```

For more information about how to add custom fields to the configuration screen, see [Adding and Editing Configuration Fields](admin-console-customize-config-screen).

If you use a Helm chart for your application in Replicated, your users provide any values specific to their environment from the configuration screen, rather than in the Helm chart `values.yaml` file.

> By allowing your users to provide configuration values in the admin console rather than in the Helm `values.yaml` file directly, you can control which options you expose to your users. It also makes it easier for your users to provide their inputs through a user interface, rather than having to edit a YAML file.

Reviewer: This got too wordy as I wrote the alternative, but I'd lead with the primary benefit which is a UI, and then have the "you get to control what you expose" as kind of an "also..." sort of thing (as it's less of a universal benefit than the improved user experience of using a UI). Anyways the below is in the right direction but feel free to slide/dice/iterate

```suggestion
By allowing your users to provide configuration values in the admin console rather than in a Helm `values.yaml` file directly or `--set` CLI commands, you make it easier for your users to provide their inputs through a user interface, rather than having to edit a YAML file or construct complex CLI invocations. It also allows you to control which options you expose to your users .
```

Author: You know, it was probably too wordy to begin with. Re-wrote:

>If you use a Helm chart for your application in Replicated, your users provide any values specific to their environment from the configuration screen, rather than in a Helm chart `values.yaml` file.
>
>This means that your users can provide configuration values through a user interface, rather than having to edit a YAML file or use `--set` CLI commands. The admin console configuration screen also allows you to control which options you expose to your users.

> For example, you can use the configuration screen to provide database configuration options for your application. Your users could connect your application to an external database by providing required values in the configuration screen, such as the host, port, and a username and password for the database. For a tutorial of this use case, see [Adding Database Configuration Options for your Application](tutorial-adding-db-config).

Reviewer: ```suggestion
For example, you can use the configuration screen to provide database configuration options for your application. Your users could connect your application to an external database by providing required values in the configuration screen, such as the host, port, and a username and password for the database. Replicated's config screen also allows you to provide an alternative batteries-included option where the database runs in-cluster as part of the application. For a tutorial of this use case, see [Adding Database Configuration Options for your Application](tutorial-adding-db-config).
```

## Viewing the Configuration Screen

Users of your application can access the configuration screen from the admin console:
* During application installation if you included required fields. See [Application Installation](#application-installation) below.
* At any time after application installation on the Config tab. See [Admin Console Config Tab](#admin-console-config-tab) below.

### Application Installation

> If you add any required fields, the admin console displays the configuration screen when the user installs the application, after they upload their license file.

Reviewer: ```suggestion
If you add a KOTS `kind: Config` object to your application manifests, the admin console displays the configuration screen when the user installs the application, after they upload their license file.
```

![configuration screen that displays during application install](../../static/images/initial-config.png)

### Admin Console Config Tab

> For applications that do not have any required fields, the admin console does not display the configuration screen during application installation.

Reviewer: False, AFAIK the config screen is always shown during setup. I will just make the one comment here and not call it out everywhere I see it from here on out.

```suggestion
```

Author: Ahh got it, ok. I'll re-work this section to reflect that
